Refonte totale pour passage Python3

kugan49 opened the pull request:
J'adore cette extension mais comme elle n'avait pas été migrée pour python 3 j'en ai profité pour la refondre complètement :

respect des PEP et des préconisation de Kodi sur la forme
je me suis inspiré (un peu) de l'add-on trakt pour l'organisation des modules
Dites moi si ca vous intéresse de conserver ce projet, ou je peux demander dans le site betaseries pour le reprendre sinon

(je n'avais pas utilisé la bonne source)

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
Add zip files

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
Kodi 18 compatibility & bi-directional synchronisation

PCCV committed:
add compatibility with netflix and catchuptv addon

PCCV committed:
add news to addons.xml

PCCV committed:
add news to addons.xml

PCCV committed:
add news to addons.xml

PCCV committed:
Create index.html

PCCV committed:
Update addons.xml

kugan49 committed:
Refonte complete

kugan49 committed:
Merge branch 'master' of https://github.com/kugan49/Kodi-betaseries

# Conflicts:
#	.gitignore
#	README.md

kugan49 committed:
Ajout du zip

skylex commented:
En effet, le code a bien changé ! :)
Par contre ton dépôt ne contient que le code du plugin "BetaSeries Agent", donc la merge request irait écraser les autres plugins (repository et subtitles).
De plus, tu as changé l'id du plugin "service.betaseries.com" en "script.betaseries", ce qui ferait que les mises à jour automatiques ne pourront pas se faire pour les personnes utilisant le plugin actuel.
Peux-tu modifier cela ?

kugan49 committed:
Formatage pour merge

kugan49 commented:
J'avais tout enlevé et renommé l'addon car j'avais envisagé de demander d'ajouter juste cet addon dans le repository officiel kodi.

C'est corrigé

PCCV commented:
Hello Kugan49,
J'avais prévu de migrer en python3 mais tu m'as coupé l'herbe sous le pied... ceci dit, je n'aurais pas été aussi loin avec les préconisations. 
J'ai voulu tester avec Kodi 19, mais j'ai un échec d'installation avec le fichier "service.betaseries.com-2.0.0" pour cause de structure invalide.
Saurais-tu à quoi cela est dû?

En tout cas, beau travail et bel effort.

kugan49 committed:
Modif structure fichier (oubli balise fermente)

kugan49 commented:
Salut, désolé j'avais oublié une baliser dans le xml des addons.
Bon tests

PCCV commented:
Bonjour kugan49,

Voici mes premiers retours :
 1 - Pour que l'on puisse faire la mise à jour à partir de la version précédente sans entrer à nouveau les identifiants et en conservant les options déjà choisies, il faudrait renommer le service de "script.betaseries" à "service.betaseries.com" dans "addon.xml", "default.py", "kodiUtilities.py" "player.py" et "utilities.py" 

 2 - Dans "serviceapi.py" 
ligne 186:
changer (showtitle.encode("ascii", "xmlcharrefreplace")).replace(" ", "+")
par 	showtitle.replace(" ", "+").encode("ascii", "xmlcharrefreplace")

et lignes 368 & 395:
changer showtitle.encode("ascii", "xmlcharrefreplace").replace(" ", "+")
par 	showtitle.replace(" ", "+").encode("ascii", "xmlcharrefreplace")

 3 - Je n'ai pas regardé pourquoi mais dans le log j'ai 3 fois:
	[script.betaseries] resources.lib.player: Scan finished, all episodes updated

Je continue à tester.
j'avais commencé à modifier l'addon pour qu'il soit compatible avec Netflix et Disney+, je vais voir ce qu'il reste a faire pour cela.

kugan49 committed:
1 - Pour que l'on puisse faire la mise à jour à partir de la version précédente sans entrer à nouveau les identifiants et en conservant les options déjà choisies, il faudrait renommer le service de "script.betaseries" à "service.betaseries.com" dans "addon.xml", "default.py", "kodiUtilities.py" "player.py" et "utilities.py"

2 - Dans "serviceapi.py"
ligne 186:
changer (showtitle.encode("ascii", "xmlcharrefreplace")).replace(" ", "+")
par showtitle.replace(" ", "+").encode("ascii", "xmlcharrefreplace")

et lignes 368 & 395:
changer showtitle.encode("ascii", "xmlcharrefreplace").replace(" ", "+")
par showtitle.replace(" ", "+").encode("ascii", "xmlcharrefreplace")

kugan49 committed:
épuration settings.xml

kugan49 commented:
Salut,

j'ai corrigé les 2 premiers points.
Pour le point 3 : je ne vois pas pourquoi tu l'as 3 fois la fonction est appelé au chargement du service et sur notification de kodi : "VideoLibrary.OnScanFinished", tu aurais 2 fois cette notification ?

Si tu veux qu'on travaille ensemble pour Netflix et Disney+ redis moi

PCCV commented:
> Pour le point 3 : je ne vois pas pourquoi tu l'as 3 fois la fonction est appelé au chargement du service et sur notification de kodi : "VideoLibrary.OnScanFinished", tu aurais 2 fois cette notification ?
J'ai installé watchdog avec les option "clean on startup" et "Scan on startup" est-ce que cela pourrait venir là ?

> Si tu veux qu'on travaille ensemble pour Netflix et Disney+ redis moi
Je vais reprendre ce que j'avais commencé pour Netflix et voir pour Disney+.
J'aimerai aussi que cela puisse fonctionner avec le replay de catchupTV mais cela dépend bcp des chaines. je te proposerai qqchose. 

En tout cas ton code est super clean,  c'est du beau travail.

kugan49 commented:
Merci, ca fait plaisir que tu t'y retrouves. 
Pour les 3 logs c'st bien watchdog puisqu'il scan en continu et une fois terminé kodi lance une notification "VideoLibrary.OnScanFinished"

On est près pour merge alors ?

PCCV commented:
Bonjour,

> On est près pour merge alors ?

Ca y est, c'est mergé. 
J'ai galéré hier (ça se voit au nb de commit): 
- pour pouvoir proposer les versions pour kodi18 et kodi19 sur le meme repo. 
- avoir accès au fichier md5 en raw, github ne me donnait accès qu'au blob!
- j'ai refait le ZIP: il faut le rep de base "service.betaseries.com"
- j'ai fait une branche pour la version précédente

Il faudra vérifier que tout fonctionne pour l'installation ainsi que pour la MàJ.

skylex commented:
Bravo c'est très propre comme code !
Il reste juste une dernière chose à changer pour que je puisse merge :
remplacer les URL https://raw.github.com/PCCV pour rester sur le repo actuel.

PCCV commented:
> remplacer les URL https://raw.github.com/PCCV pour rester sur le repo actuel.

j'ai fait une branche pour Kodi18, je laisse les liens ou je les change aussi ?

skylex commented:
Oui pareil, je vais créer la branche leia pour que tu puisses faire une merge request dessus.
Merci

jaumard commented:
Hey :) du coup il faut installer quoi et ou? un petite doc serais top 
Car la je vois que rien est merge ni sur quel repo je dois récupérer l'addon. J'ai essayé d'installé https://github.com/PCCV/xbmc-betaseries branche master mais je me retrouve avec l'écran des settings sans aucun texte juste les checkbox :D et rien pour télécharger les sous titres. 

Merci merci :)

rmathot commented:
Bonjour, est-ce que je peux filer un coup de main niveau code/testing/debugging pour faire avancer cette PR? Comme j'aime bien cette extension, j'ai un peu de temps à donner :)
